skrub/datasets/_fetching.py

```diff
@@ -702,6 +703,8 @@ def _fetch_dataset_as_dataclass(
             df = pd.read_parquet(info["path"])
         else:
             df = pd.read_csv(info["path"], **read_csv_kwargs)
+        if shuffling:
+            df = shuffle(df, random_state=42).reset_index(drop=True)
```

Comment:
I think the random state should be a function parameter, as it would allow for control for multiple seeds during testing and more fine-grained control on reproducibility.

I'm not 100% sure we should reset indices, because it could be surprising for users. WDYT?